src/modules/commander/failure_detector/FailureDetector.cpp

```diff
@@ -47,50 +47,42 @@ FailureDetector::FailureDetector(ModuleParams *parent) :
 {
 }
 
-bool FailureDetector::resetAttitudeStatus()
+void FailureDetector::resetAttitudeStatus()
 {
-
 	int attitude_fields_bitmask = _status & (FAILURE_ROLL | FAILURE_PITCH | FAILURE_ALT | FAILURE_EXT);
-	bool status_changed(false);
 
 	if (attitude_fields_bitmask > FAILURE_NONE) {
 		_status &= ~attitude_fields_bitmask;
-		status_changed = true;
 	}
-
-	return status_changed;
 }
 
-bool
-FailureDetector::update(const vehicle_status_s &vehicle_status)
+bool FailureDetector::update(const vehicle_status_s &vehicle_status)
 {
-
-	bool updated(false);
+	uint8_t previous_status = _status;
 
 	if (isAttitudeStabilized(vehicle_status)) {
-		updated |= updateAttitudeStatus();
+		updateAttitudeStatus();
 
 		if (_param_fd_ext_ats_en.get()) {
-			updated |= updateExternalAtsStatus();
+			updateExternalAtsStatus();
```

Comment:
Why should "external ATS" be dependent on attitude stabilized?
@dakejahl

Reply:
When flying manually in attitude or rates mode you probably don't want your ATS device to trigger the parachute, since you are in manual control anyway and you are probably doing some sketchy maneuvers (as pilots like to do).

Reply:
I could see it go either way though

Reply:
Ok, I thought this was simply telling you the external thing triggered a parachute and you should now shut off your motors so it actually works?

Either way maybe we should add a comment.

Reply:
Oh. Yeah you're right, that's totally how it works. Yes it should be changed.

Reply:
I didn't change that logic. Should I? Can we do a follow-up pr?

Reply:
Whatever's easier.

Reply:
Let's do a follow up to not derail.